From 7362afd02da514072acf8e461062013ab9e34fd5 Mon Sep 17 00:00:00 2001
From: Christian Hohnstaedt <Christian.Hohnstaedt@wago.com>
Date: Thu, 14 May 2020 12:21:49 +0200
Subject: [PATCH] Enable PTN5150 device driver and adjust the device tree

The enable pin ( (&io_expander_70 6) for the PTN5150
is currently ignored. (High-Z input)

The PTN5150 is turned on by the voltage divider connected to ENB.

Signed-off-by: Christian Hohnstaedt <Christian.Hohnstaedt@wago.com>
---
 arch/arm/boot/dts/am335x-pfc-768_3301.dts         |  6 +++++-
 arch/arm/boot/dts/am335x-pfc-768_330x.dtsi        | 14 +++++++++++++-
 arch/arm/configs/am335x_pfc_adv_generic_defconfig |  2 ++
 3 files changed, 20 insertions(+), 2 deletions(-)

diff --git a/arch/arm/boot/dts/am335x-pfc-768_3301.dts b/arch/arm/boot/dts/am335x-pfc-768_3301.dts
index 09510251e80a..d38afb36e39e 100644
--- a/arch/arm/boot/dts/am335x-pfc-768_3301.dts
+++ b/arch/arm/boot/dts/am335x-pfc-768_3301.dts
@@ -63,7 +63,7 @@
 };
 
 &usb0 {
-	dr_mode = "peripheral";
+	dr_mode = "otg";
 	status = "okay";
 };
 
@@ -147,3 +147,7 @@
 		spi-max-frequency = <48000000>;
 	};
 };
+
+&cc_logic_usb_c {
+	status = "okay";
+};
diff --git a/arch/arm/boot/dts/am335x-pfc-768_330x.dtsi b/arch/arm/boot/dts/am335x-pfc-768_330x.dtsi
index 0708fb57ffd7..7b95321ce348 100644
--- a/arch/arm/boot/dts/am335x-pfc-768_330x.dtsi
+++ b/arch/arm/boot/dts/am335x-pfc-768_330x.dtsi
@@ -149,7 +149,14 @@
 	gpio0_pins: pinmux_gpio0_pins {
 		pinctrl-single,pins = <
 			AM33XX_XDMA_EVENT_INTR0_GPIO0_19(PIN_INPUT) /* nRST-LEDServer */
+		>;
+	};
+
+	ptn5150_pins: pinmux_ptn5150_pins {
+		pinctrl-single,pins = <
 			AM33XX_GPMC_WPN_GPIO0_31(PIN_INPUT) /* nINT-MCU-CC */
+			/* USB0 VBUS activate */
+			AM33XX_USB0_DRVVBUS_GPIO0_18(PIN_OUTPUT_PULLDOWN)
 		>;
 	};
 
@@ -225,7 +232,12 @@
 	 * Pins: nRST-CC (&io_expander_70 6)  and nINT-CC
 	 */
 	cc_logic_usb_c: cclogic@3c {
-		reg = <0x3c>;
+		compatible = "nxp,ptn5150";
+		reg = <0x1e>;
+		int-gpio = <&gpio0 31 GPIO_ACTIVE_HIGH>;
+		vbus-gpio = <&gpio0 18 GPIO_ACTIVE_HIGH>;
+		pinctrl-names = "default";
+		pinctrl-0 = <&ptn5150_pins>;
 	};
 };
 
diff --git a/arch/arm/configs/am335x_pfc_adv_generic_defconfig b/arch/arm/configs/am335x_pfc_adv_generic_defconfig
index 5c621c43df8d..f5f2d18a9b79 100644
--- a/arch/arm/configs/am335x_pfc_adv_generic_defconfig
+++ b/arch/arm/configs/am335x_pfc_adv_generic_defconfig
@@ -446,6 +446,8 @@ CONFIG_UIO_PDRV_GENIRQ=y
 CONFIG_STAGING=y
 CONFIG_HWSPINLOCK_OMAP=m
 # CONFIG_IOMMU_SUPPORT is not set
+CONFIG_EXTCON=m
+CONFIG_EXTCON_PTN5150=m
 CONFIG_IIO=y
 CONFIG_IIO_BUFFER_CB=y
 CONFIG_TI_AM335X_ADC=y
-- 
2.17.1

